edx_exams/apps/router/tests/test_views.py

```diff
@@ -0,0 +1,211 @@
+        Verify the endpoint validates permissions
+        """
+
+        # Test unauthenticated
+        response = self.client.patch(self.url)
+        self.assertEqual(response.status_code, 401)
+
+        # Test non-staff worker
+        random_user = UserFactory()
+        response = self.patch_api(random_user, [])
+        self.assertEqual(response.status_code, 403)
+
+    def patch_api(self, user, data):
+        """
+        Helper function to make a patch request to the API
+        """
+        data = json.dumps(data)
+        headers = self.build_jwt_headers(user)
+        return self.client.patch(self.url, data, **headers, content_type="application/json")
+
+    def build_mock_response(self):
+        mock_response = Mock(spec=Response)
+        mock_response.json.return_value = {}
+        mock_response.status_code = 200
+        return mock_response
+
+    @mock.patch('edx_exams.apps.router.views.register_exams')
+    def test_patch_exams(self, mock_register_exams):
+        """
+        Request is forwarded to the LMS with exam type
+        broken into individual attributes
+        """
+        data = [
+            {
+                'course_id': self.course_id,
+                'content_id': '123foo',
+                'exam_name': 'test practice exam',
+                'exam_type': 'practice',
+                'backend': 'mockprock'
+            },
+            {
+                'course_id': self.course_id,
+                'content_id': '123bar',
+                'exam_name': 'test proctored exam',
+                'exam_type': 'proctored',
+                'backend': 'mockprock'
+            },
+        ]
+
+        mock_register_exams.return_value = self.build_mock_response()
+        response = self.patch_api(self.user, data)
+        self.assertEqual(response.status_code, 200)
+
+        mock_register_exams.assert_called_with(
+            self.course_id,
+            [
+                {
+                    'course_id': self.course_id,
+                    'content_id': '123foo',
+                    'exam_name': 'test practice exam',
+                    'exam_type': 'practice',
+                    'backend': 'mockprock',
+                    'is_proctored': False,
+                    'is_practice_exam': True,
+                },
+                {
+                    'course_id': self.course_id,
+                    'content_id': '123bar',
+                    'exam_name': 'test proctored exam',
+                    'exam_type': 'proctored',
+                    'backend': 'mockprock',
+                    'is_proctored': True,
+                    'is_practice_exam': False,
+                },
+            ]
+        )
+        self.assertEqual(Exam.objects.all().count(), 1)
+
+    @mock.patch('edx_exams.apps.router.views.register_exams')
+    def test_patch_exams_empty(self, mock_register_exams):
+        """
+        Request is forwarded to the LMS, no exams in this
+        service should be removed.
```

Comment:
Question: could you clarify why exams could/would be removed? In other words, what is being tested here? Thanks!

Reply:
updated the comment a bit but the patch endpoint takes all of the exams in the course and syncs to that. So if studio sends an empty list, it means all exams would be removed.